src/zenml/integrations/kubeflow/orchestrators/kubeflow_dag_runner.py

```diff
@@ -350,12 +348,7 @@ def _construct_pipeline_graph(
             # remove the extra pipeline node information
             tfx_node_ir = self._dehydrate_tfx_ir(tfx_ir, component.id)
 
-            from zenml.utils import source_utils
-
-            main_module_file = cast(str, sys.modules["__main__"].__file__)
-            main_module = source_utils.get_module_source_from_file_path(
-                os.path.abspath(main_module_file)
-            )
+            main_module = sys.modules["__main__"].__name__
```

Comment:
I think the name of the main module will always be `__main__`. Let me explain why this matters when passing it to kubeflow: Inside the kubeflow entrypoint, we try to import the user main file in order to also import any materializers or required custom code that got imported when the pipeline is run locally. When we're passing `__main__`, the entrypoint won't know which file to import (and in case the step is in the same file, won't be able to find the step implementation either). Maybe there is some detail that I'm missing as I haven't quite finished the review, but it seems like this should fail for many cases (same with the step operators entrypoint)

Reply:
This is something that Kubeflow integration test cases also caught. I switched everything back to how it was (more or less), but I still allow code to run outside of a .zen repository, in which case the "root" is considered to be the folder where
the main module file is located.